pkg/reconciler/brokercell/controller.go

```diff
@@ -81,6 +108,8 @@ func NewController(
 	endpointsinformer.Get(ctx).Informer().AddEventHandler(handleResourceUpdate(impl))
 	// 3. Watch hpa for ingress, fanout and retry deployments
 	hpainformer.Get(ctx).Informer().AddEventHandler(handleResourceUpdate(impl))
+	// 4. Watch the broker targets configmap.
+	configmapinformer.Get(ctx).Informer().AddEventHandler(handleResourceUpdate(impl))
```

Comment:
The reason was exactly the Endpoints - it doesn't inherit the owner reference. I decided to use the same label based approach for all resources. Alternatively we can filter by owner for other resources, and use label for endpoints.

Reply:
I have a minor preference for using owner reference where possible, just because that's the convention we use in other controllers that directly create objects. But doesn't have to be changed in this PR, and I don't mind if you decide not to change it.

Reply:
SG. I will do this in a follow up then since this is not specific to this PR and this PR is already pretty big.